py/server/tests/test_udf_return_java_values.py

```diff
@@ -290,9 +297,30 @@ def test_np_ufunc(self):
         @numba.vectorize([numba.float64(numba.float64)], nopython=True)
         def nbsin(x):
             return np.sin(x)
+
         t3 = empty_table(10).update(["X3 = nbsin(i)"])
         self.assertEqual(t3.columns[0].data_type, dtypes.double)
 
+    def test_md_np_array(self):
+        np_1d_array = np.array([1, 2, 3, 4, 5, 6, 7, 8], dtype=np.int64)
+        np_2d_array = np_1d_array.reshape((4, 2))
+        np_3d_array = np_1d_array.reshape((2, 2, 2))
+
+        def f(ndim):
+            if ndim == 1:
+                return np_1d_array
+            elif ndim == 2:
+                return np_2d_array
+            elif ndim == 3:
+                return np_3d_array
+            else:
+                raise ValueError("ndim must be 1, 2, or 3")
+
+        for ndim in [1, 2, 3]:
+            with self.subTest(ndim):
+                cast_str = f"(long{'[]' * ndim})"
+                t = empty_table(10).update([f"X1 = {cast_str}f(ndim)"])
+                self.assertEqual(t.columns[0].data_type, dtypes.from_jtype(jpy.get_type("[" * ndim + "J").jclass))
```

Comment:
What happens if `f()` returns different dimensioned results?  Do we get an error?  Does it "work" but do weird stuff?  I'd check this case.